src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -0,0 +1,86 @@
+export class FilterTimestamp extends HTMLElement {
+  constructor() {
+    super();
+
+    this.timeStamp = {};
+    this.onHiddenListener = () => browser.menus.removeAll();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const filterTSWrapper = document
+      .querySelector('#filterTimestampTemplate')
+      .content.cloneNode(true);
+
+    this.filterTSContainer = filterTSWrapper.querySelector(
+      '.timestamp-filter-container'
+    );
+
+    this.startTimeLabel = filterTSWrapper.querySelector('#startTimeLabel');
+    this.stopTimeLabel = filterTSWrapper.querySelector('#stopTimeLabel');
+    this.clearTSFilterBtn = filterTSWrapper.querySelector('#clearBtn');
+
+    shadow.appendChild(filterTSWrapper);
+  }
+
+  setFilterRange = (info, tab) => {
+    if (tab.id !== this.tab.id) {
```

Comment:
Why are you checking the tab here?

The `getTargetElement` call below ensures that you're only handling events from this page.

If you want to handle multiple menu items, check `info.menuItemId` to see which menu item had been clicked.